omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java

```diff
@@ -62,6 +81,75 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         }
     }
 
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        String requestURI = request.getRequestURI();
+        if (requestURI.endsWith("/config.json")) {
+            if (!Context.isAuthenticated()) {
+                String basicAuth = request.getHeader("Authorization");
+                if (basicAuth != null) {
+                    // check that header is in format "Basic ${base64encode(username + ":" + password)}"
+                    if (basicAuth.startsWith("Basic")) {
+                        try {
```

Comment:
Can we simplify the `doPost` by wrapping this in private util method/s? This method could still be in the same class.